tools/ports/sdl2_ttf.py

```diff
@@ -5,22 +5,22 @@
 
 import os
 
-TAG = 'version_1'
-HASH = '6ce426de0411ba51dd307027c4ef00ff3de4ee396018e524265970039132ab20adb29c2d2e61576c393056374f03fd148dd96f0c4abf8dcee51853dd32f0778f'
+TAG = '703ebc7c66fd'  #  Latest as of 24 November 2020
+HASH = '362c95f08170a90fe026b264a1053692edea581f5a9908c84aa42d89315cf8c8f74719a80ae45fa16ed7944b12e1b930e5add1d3ad58de04012ed63f535dfb04'
 
-deps = ['freetype', 'sdl2']
+deps = ['freetype', 'sdl2', 'harfbuzz']
 
 
 def needed(settings):
   return settings.USE_SDL_TTF == 2
 
 
 def get(ports, settings, shared):
-  ports.fetch_project('sdl2_ttf', 'https://github.com/emscripten-ports/SDL2_ttf/archive/' + TAG + '.zip', 'SDL2_ttf-' + TAG, sha512hash=HASH)
+  ports.fetch_project('sdl2_ttf','https://hg.libsdl.org/SDL_ttf/archive/' + TAG + '.zip', 'SDL2_ttf-' + TAG, sha512hash=HASH)
   libname = ports.get_lib_name('libSDL2_ttf')
 
   def create():
-    src_root = os.path.join(ports.get_dir(), 'sdl2_ttf', 'SDL2_ttf-' + TAG)
+    src_root = os.path.join(ports.get_dir(), 'sdl2_ttf', 'SDL_ttf-' + TAG)
```

Comment:
This name change may explain the error on CI. I think you need to update `embuilder.py` as well.

Reply:
@kripken thanks for pointing it out. Is there anyway I can run the tests locally? Any hints would go a long way.

Reply:
I installed the requirements from the requirments file and I tried `pytest tests` but I get many many errors. So I think the env is not properly set. I also checked the circle pi file but I didn't mange to replicate it locally.

Reply:
You should be able to run the test locally with `./tests/runner.py browser.test_sdl2_ttf`. But the part that fails is the test for the frozen cache. You can set that in your `.emscripten` file, but then you'd also need to do the other stuff the CI does.

Maybe the simplest thing is to
* Do `emcc --clear-cache --clear-ports`
* Run the embuider command that CI does.
* Run that test, and see if it mentions it builds anything - any building would be an error.

Reply:
(looks like the embuilder command is `./embuilder.py build ALL`)

Reply:
@kripken thanks for the tips. I was able to build and run the tests as you said, however I got a strange failure in sdl2_ttf test which I can't undersand. Perhaps you have a better idea: 
```AssertionError: Unexpected difference:
--- expected
+++ actual
@@ -1 +1 @@
-/report_result?0
+/report_result?12```

Moreover, regarding the SDL2_ttf to SDL_ttf name change, I noticed the zipfile downloaded from the upstream contains a folder named `SDL_ttf` and extracts to a folder named accordingly, hence using `SDL_ttf` in the script.